sechub-wrapper-owasp-zap/src/test/java/com/mercedesbenz/sechub/zapwrapper/util/TOTPGeneratorTest.java

```diff
@@ -1,42 +1,42 @@
 // SPDX-License-Identifier: MIT
 package com.mercedesbenz.sechub.zapwrapper.util;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.*;
 
 import java.nio.charset.StandardCharsets;
 import java.util.stream.Stream;
 
 import org.apache.commons.codec.DecoderException;
 import org.apache.commons.codec.binary.Base32;
-import org.apache.commons.codec.binary.Hex;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.ArgumentsProvider;
 import org.junit.jupiter.params.provider.ArgumentsSource;
 
+import com.mercedesbenz.sechub.commons.model.login.EncodingType;
 import com.mercedesbenz.sechub.commons.model.login.TOTPHashAlgorithm;
+import com.mercedesbenz.sechub.commons.model.login.WebLoginTOTPConfiguration;
 
 class TOTPGeneratorTest {
 
     @Test
     void secret_key_being_null_throws_exception() {
         /* execute + test */
-        assertThrows(IllegalArgumentException.class, () -> new TOTPGenerator(null, 6, TOTPHashAlgorithm.HMAC_SHA1, 30));
+        assertThrows(NullPointerException.class, () -> new TOTPGenerator(new WebLoginTOTPConfiguration()));
```

Comment:
Please check the content of the NPE here. So it is clear it is not another (unwanted) NPE but the thing we assert...

or we use the UserInfoScriptException here .... and we could here then check the content of the message